game/src/game/components/VehicleComponent.cpp

```diff
@@ -135,10 +142,19 @@ std::string_view VehicleComponent::GetName() const
 
 void VehicleComponent::SetVehicleName(const string& vehicle_name)
 {
-    const PxTransform pose(GlmToPx(transform_->GetPosition()),
-                           PxQuat(PxIdentity));
+    physx::PxQuat quat(
+        transform_->GetOrientation().x, transform_->GetOrientation().y,
+        transform_->GetOrientation().z, transform_->GetOrientation().w);
+
+    PxTransform pose(GlmToPx(transform_->GetPosition()), quat);
 
     g_vehicle_name_ = vehicle_name;
     g_vehicle_.setUpActor(*physics_service_->GetKScene(), pose,
                           g_vehicle_name_.c_str());
+
+    const physx::PxVec3 pose_transform =
+        g_vehicle_.mPhysXState.physxActor.rigidBody->getGlobalPose().p;
+
+    g_vehicle_.mPhysXState.physxActor.rigidBody->setGlobalPose(
+        physx::PxTransform(pose_transform, quat));
```

Comment:
I'm not sure I understand why this is being done? From what I remember, `setUpActor()` already does set the global pose on the rigidbody to whatever you give it, so why get the global pose and set it again?

Also, why not use the `CreatePxTransform` or `GlmToPx`/`PxToGlm` utils?